docs/rfcs/1000-shamir-based-recovery.md

```diff
@@ -730,6 +731,121 @@ The claimer gets access to `reveal_token` and `data_key`, it can then retrieve `
 Then `ciphered_data` can be decrypted with `data_key`. From then on, the recovery works
 just like the recovery device system (see `parsec core import_recovery_device` CLI).
 
+## 4 - Delete a shamir setup
+
+Authenticated API:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "shamir_recovery_deletion",
+            "fields": [
+                {
+                    "name": "certificate",
+                    // Contains a ShamirRecoveryDeletionCertificate
+                    "type": "bytes"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok"
+            },
+            {
+                // Cannot deserialize data into the expected certificate
+                "status": "invalid_data"
+            },
+            {
+                // Given timestamp does not match previous timestamps
+                "status": "incoherent_previous_timestamp"
+            }
+            {
+                // Nothing to delete
+                "status": "no_setup_to_delete"
```

Comment:
```suggestion
                "status": "shamir_setup_not_found"
```

to stay consistent with e.g. [`user_not_found`](https://github.com/Scille/parsec-cloud/blob/048b4c6528cbb1b78b6daa9d567f356c6307462f/libparsec/crates/protocol/schema/authenticated_cmds/user_revoke.json5#L23) and [`realm_not_found`](https://github.com/Scille/parsec-cloud/blob/048b4c6528cbb1b78b6daa9d567f356c6307462f/libparsec/crates/protocol/schema/authenticated_cmds/realm_share.json5#L33
And `shamir_setup` instead of just `shamir` given it's how it is specified in `shamir_recovery_setup`:
https://github.com/Scille/parsec-cloud/blob/048b4c6528cbb1b78b6daa9d567f356c6307462f/libparsec/crates/protocol/schema/authenticated_cmds/shamir_recovery_setup.json5#L54

I'm also okay to stick with just `setup`, so the vocabulary would be "a `ShamirRecoveryBrief` describes a setup, and has been added in the server with the `shamir_recovery_setup` command", but in this case we have to correct the `shamir_recovery_setup` command and make sure we are consistent everywhere (for instance I didn't check the comments in the code)

Reply:
On top of that I would add a `shamir_setup_already_deleted` (with a `DateTime` field `last_shamir_recovery_certificate`) to cover the case the shamir setup exists but can no longer be deleted (see the comment about `shamir_setup_already_exists` for the reason for the field `last_shamir_recovery_certificate`)